File: android-tv-smart-triggers/app/src/main/java/com/nexuscast/player/BootReceiver.java
package com.nexuscast.player;

  import android.content.BroadcastReceiver;
  import android.content.Context;
  import android.content.Intent;
  import android.content.SharedPreferences;
  import android.os.Build;

  /**
   * Listens for device boot and starts BootLaunchService to open MainActivity.
   * Android 10+ / FireOS 7+ silently drops background startActivity() for
   * non-system apps. A foreground service bypasses this restriction.
   * LOCKED_BOOT_COMPLETED fires before credential unlock; skip prefs check there.
   */
  public class BootReceiver extends BroadcastReceiver {

      private static final String PREFS_NAME = "DigipalPrefs";
      private static final String KEY_AUTO_RELAUNCH = "auto_relaunch";

      @Override
      public void onReceive(Context context, Intent intent) {
          String action = intent.getAction();
          if (action == null) return;

          boolean isLockedBoot = "android.intent.action.LOCKED_BOOT_COMPLETED".equals(action);
          boolean isStandardBoot = Intent.ACTION_BOOT_COMPLETED.equals(action)
                  || "android.intent.action.QUICKBOOT_POWERON".equals(action);
          if (!isLockedBoot && !isStandardBoot) return;

          if (isStandardBoot) {
              try {
                  SharedPreferences prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
                  if (!prefs.getBoolean(KEY_AUTO_RELAUNCH, false)) return;
              } catch (Exception e) { return; }
          }

          Intent svc = new Intent(context, BootLaunchService.class);
          try {
              if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) context.startForegroundService(svc);
              else context.startService(svc);
          } catch (Exception e) {
              try {
                  Intent launch = new Intent(context, MainActivity.class);
                  launch.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
                  context.startActivity(launch);
              } catch (Exception ignored) {}
          }
      }
  }
  

File: vega/android/app/src/main/java/com/digipalvega/BootReceiver.kt
package com.digipalvega

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import androidx.core.content.ContextCompat

/**
 * Auto-launches the player on device boot.
 *
 * Handles three boot intents:
 *  - BOOT_COMPLETED          — standard boot (all Android versions)
 *  - LOCKED_BOOT_COMPLETED   — direct-boot mode (FireOS 8 fires this early in boot)
 *  - QUICKBOOT_POWERON       — HTC/Amazon fast-boot variant
 *
 * Why BootLaunchService instead of startActivity() directly:
 * Android 10+ (API 29) and FireOS 7/8 block background startActivity() for
 * non-system apps — the broadcast fires but the activity start is silently
 * dropped. Starting a foreground service from a receiver IS allowed, and the
 * service can then open the activity from a foreground context.
 */
class BootReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action ?: return
        if (action != Intent.ACTION_BOOT_COMPLETED &&
            action != "android.intent.action.LOCKED_BOOT_COMPLETED" &&
            action != "android.intent.action.QUICKBOOT_POWERON") {
            return
        }

        val serviceIntent = Intent(context, BootLaunchService::class.java)
        try {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                context.startForegroundService(serviceIntent)
            } else {
                context.startService(serviceIntent)
            }
        } catch (e: Exception) {
            // Fallback: try direct activity start (may silently fail on API 29+
            // but better than nothing if the service itself fails to start).
            try {
                val launch = Intent(context, MainActivity::class.java).apply {
                    addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
                    addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
                }
                context.startActivity(launch)
            } catch (e2: Exception) {
                // Nothing more we can do.
            }
        }
    }
}
